packages/grafana-prometheus/src/datasource.ts

```diff
@@ -471,7 +471,9 @@ export class PrometheusDatasource
 
     // Use incremental query only if enabled and no instant queries or no $__range variables
     const shouldUseIncrementalQuery =
-      this.hasIncrementalQuery && !request.targets.some((target) => target.instant || target.expr.includes('$__range'));
+      this.hasIncrementalQuery &&
+      !request.targets.some((target) => target.instant || target.expr?.includes('$__range')) &&
+      !config.publicDashboardAccessToken;
```

Comment:
I'd move this one line up to fail the check quickly. If there is a token it's going to fail anyways. No need to wait for `some` loop.